src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,211 @@
+	 */
+	protected function get_post_suggestions( $search, $per_page, $offset = 0 ): array {
+		$post_suggestions    = [];
+		$excluded_post_types = [ 'attachment' ];
+
+		$post_types = $this->wp->get_post_types(
+			[
+				'exclude_from_search' => false,
+				'public'              => true,
+			]
+		);
+
+		// Exclude attachment post_type
+		$filtered_post_types = array_diff( $post_types, $excluded_post_types );
+
+		$args = [
+			'post_type'      => $filtered_post_types,
+			'posts_per_page' => $per_page,
+			'post_status'    => 'publish',
+			's'              => $search,
+			'offset'         => $offset,
+		];
+
+		$posts = $this->wp->get_posts( $args );
+
+		foreach ( $posts as $post ) {
+			$post_suggestions[] = $this->format_final_url_response( $post->ID, 'post', $post->post_title, get_permalink( $post->ID ) );
+		}
+
+		return $post_suggestions;
+	}
+
+	/**
+	 * Get terms that can be used to suggest assets
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 *
+	 * @return array
+	 */
+	protected function get_terms_suggestion( $search, $per_page ): array {
+		$terms_suggestions = [];
+
+		// Get all taxonomies that are public, show_in_menu = true helps to exclude taxonomies such as "product_shipping_class".
+		$taxonomies = $this->wp->get_taxonomies(
+			[
+				'public'       => true,
+				'show_in_menu' => true,
+			],
+		);
+
+		$terms = $this->wp->get_terms(
+			[
+				'taxonomy'   => $taxonomies,
+				'hide_empty' => false,
+				'number'     => $per_page,
+				'name__like' => $search,
+			]
+		);
+
+		foreach ( $terms as $term ) {
+				$terms_suggestions[] = $this->format_final_url_response( $term->term_id, 'term', $term->name, get_term_link( $term->term_id, $term->taxonomy ) );
+		}
+
+		return $terms_suggestions;
+
+	}
+
+
+	/**
+	 * Return a list of final urls that can be used to suggest assets.
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 * @param string $order_by Order by: id, type, title, url
+	 *
+	 * @return array final urls with their title, id & type.
+	 */
+	public function get_final_urls_suggestions( $search = '', $per_page = 30, $order_by = 'title' ): array {
+		if ( empty( $search ) ) {
+			 return $this->get_defaults_final_urls_suggestions();
+		}
+
+		// Split possible results between posts and terms.
+		$per_page_posts = (int) ceil( $per_page / 2 );
+
+		$posts = $this->get_post_suggestions( $search, $per_page_posts );
+
+		// Try to get more results using the terms
+		$per_page_terms = $per_page - count( $posts );
+
+		$terms = [];
+
+		// get_terms  evaluates $per_page_terms = 0 as a falsy, therefore it will not add the LIMIT clausure returning all the results.
+		// See: https://github.com/WordPress/WordPress/blob/abe134c2090e84080adc46187884201a4badd649/wp-includes/class-wp-term-query.php#L868
+		if ( $per_page_terms > 0 ) {
+			$terms = $this->get_terms_suggestion( $search, $per_page_terms );
+		}
```

Comment:
💅 

Adding 

```
if ( $per_page <= 0 ) {
  return [];
}
```
Inside `get_terms_suggestion` instead of here, could reduce the verbosity of this code to just:

`$terms = $this->get_terms_suggestion( $search, $per_page_terms );`